### PR TITLE
Windup 3152

### PR DESCRIPTION
**Pull request by PhilipCattanach**
This PR is to cater for the renaming of the javax packages to jakarta in application import statements.

### CODE REVIEW

File: rules-reviewed/eap8/eap7/javax-to-jakarta-package.windup.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                        <quickfix type="REPLACE" name="javax-replace">
+                            <replacement>jakarta</replacement>
+                            <search>javax</search>
+                        </quickfix>
+                    </hint>
+                </iteration>
+            </perform>
+            <where param="renamed">
+                <matches
+                    pattern="(activation|annotation|batch|decorator|ejb|el|enterprise|faces|inject|interceptor|jms|json|jws|mail|persistence|resource|security|servlet|transaction|validation|websocket|ws|xml)"/>
```
Comment (m-brophy) [inline rules-reviewed/eap8/eap7/javax-to-jakarta-package.windup.xml:40]: Are these the jakarta ee 9 packages? my only concern is that there are javax packages in previous versions which have been removed for various reasons and these won't be flagged

Comment (PhilipCattanach) [reply rules-reviewed/eap8/eap7/javax-to-jakarta-package.windup.xml:40]: Hi Mark, The list of packages that I included in the rule are those listed in the [Eclipse Transformer package renames properties file](https://github.com/eclipse/transformer/blob/main/org.eclipse.transformer.cli/src/main/resources/org/eclipse/transformer/jakarta/jakarta-renames.properties). The goal of the rule is to rename the packages imported when moving to Jakarta EE 9. Not to highlight javax packages that have been removed in earlier versions. Cheers, Phil

File: rules-reviewed/eap8/eap7/tests/javax-to-jakarta-package.windup.test.xml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+          xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <testDataPath>data</testDataPath>
+    <rulePath>../javax-to-jakarta-package.windup.xml</rulePath>
+    <ruleset>
+        <rules>
+            <rule id="javax-to-jakarta-import-00001-test">
+                <when>
+                    <not>
+                        <iterable-filter size="23"> 
+                            <hint-exists message="Replace the `jaxax."/>
```
Comment (sshveta) [inline rules-reviewed/eap8/eap7/tests/javax-to-jakarta-package.windup.test.xml:14]: Spelling mistake in javax

File: rules-reviewed/eap8/eap7/javax-to-jakarta-package.windup.xml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            <when>
+                <javaclass references="javax.{renamed}{*}" as="javaClass">
+                    <location>IMPORT</location>
+                </javaclass>
+            </when>
+            <perform>
+                <iteration over="javaClass">
+                    <hint title="javax.{renamed} has been replaced by jakarta.{renamed}" effort="1"
+                          category-id="mandatory">
+                        <message>Replace the `jaxax.{renamed}` import statement with `jakarta.{renamed}` </message>
```
Comment (sshveta) [inline rules-reviewed/eap8/eap7/javax-to-jakarta-package.windup.xml:28]: Spelling mistake in javax

Comment (PhilipCattanach) [reply rules-reviewed/eap8/eap7/javax-to-jakarta-package.windup.xml:28]: Hi Shveta,
We managed to catch that and fix it in a subsequent PR
If you have a look at the source on Master for https://github.com/windup/windup-rulesets/blob/master/rules-reviewed/eap8/eap7/javax-to-jakarta-package.windup.xml and https://github.com/windup/windup-rulesets/blob/master/rules-reviewed/eap8/eap7/tests/javax-to-jakarta-package.windup.test.xml
You will see the correct javax 
Thanks,
Phil

**Pull request merged**